tls/extensions/s2n_server_max_fragment_length.c

```diff
@@ -24,7 +24,44 @@
 
 #include "tls/extensions/s2n_server_max_fragment_length.h"
 
-/* Precalculate size of extension */
+static bool s2n_max_fragment_length_should_send(struct s2n_connection *conn);
+static int s2n_max_fragment_length_send(struct s2n_connection *conn, struct s2n_stuffer *out);
+static int s2n_max_fragment_length_recv(struct s2n_connection *conn, struct s2n_stuffer *extension);
+
+const s2n_extension_type s2n_server_max_fragment_length_extension = {
+    .iana_value = TLS_EXTENSION_MAX_FRAG_LEN,
+    .is_response = true,
+    .send = s2n_max_fragment_length_send,
+    .recv = s2n_max_fragment_length_recv,
+    .should_send = s2n_max_fragment_length_should_send,
+    .if_missing = s2n_extension_noop_if_missing,
+};
+
+static bool s2n_max_fragment_length_should_send(struct s2n_connection *conn)
+{
+    return conn && conn->mfl_code;
+}
+
+static int s2n_max_fragment_length_send(struct s2n_connection *conn, struct s2n_stuffer *out)
```

Comment:
Why is this function not `bool` type?

Reply:
Because it can return an error and needs to match the send function pointer